addons/coupon/__manifest__.py

```diff
@@ -6,7 +6,7 @@
     'description': """Integrate coupon mechanism in orders.""",
     'category': 'Sales',
     'version': '1.0',
-    'depends': ['account'],
+    'depends': ['account', 'barcodes'],
```

Comment:
You can put the "barcode" stuff in pos_coupon, as a barcode will never be used in sale